sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/FileServiceAsyncAPITests.groovy

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.spock
+
+import com.azure.storage.common.credentials.SharedKeyCredential
+import com.azure.storage.file.ShareAsyncClient
+import com.azure.storage.file.models.CorsRule
+import com.azure.storage.file.models.FileServiceProperties
+import com.azure.storage.file.models.ListSharesOptions
+import com.azure.storage.file.models.Metrics
+import com.azure.storage.file.models.RetentionPolicy
+import com.azure.storage.file.models.ShareItem
+import com.azure.storage.file.models.ShareProperties
+import com.azure.storage.file.models.StorageErrorCode
+import reactor.test.StepVerifier
+import spock.lang.Unroll
+
+import static org.junit.Assert.assertEquals
+
+class FileServiceAsyncAPITests extends APISpec {
+    def shareName
+
+    static def testMetadata = Collections.singletonMap("testmetadata", "value")
+    static def reallyLongString = "thisisareallylongstringthatexceedsthe64characterlimitallowedoncertainproperties"
+    static def TOO_MANY_RULES = new ArrayList<>()
```

Comment:
Same as the synchronous test comments. Maybe add a helper method in APISpec that creates a valid `CorsRule` then override these properties with the invalid value.